src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentPosterImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.atlassian.bitbucket.jenkins.internal.deployments;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactory;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.AuthorizationException;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.GlobalCredentialsProvider;
+import com.atlassian.bitbucket.jenkins.internal.credentials.JenkinsToBitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketCDCapabilities;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.cloudbees.plugins.credentials.Credentials;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Singleton
+public class DeploymentPosterImpl implements DeploymentPoster {
+
+    protected static final Logger LOGGER = Logger.getLogger(DeploymentPosterImpl.class.getName());
+
+    @Inject
+    private BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+    @Inject
+    private JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials;
+    @Inject
+    private BitbucketPluginConfiguration pluginConfiguration;
+
+    @Override
+    public void postDeployment(String bbsServerId, String projectKey, String repositorySlug, String revisionSha,
+                               BitbucketDeployment deployment, Run<?, ?> run,
+                               TaskListener taskListener) {
+        Optional<BitbucketServerConfiguration> maybeServer = pluginConfiguration.getServerById(bbsServerId);
+        if (!maybeServer.isPresent()) {
+            taskListener.error(String.format("Could not send deployment notification to Bitbucket Server: Unknown serverId %s", bbsServerId));
```

Comment:
I would statically import `format` so you don't have to drag the `String` qualifier in every time